internal/component/otelcol/processor/processor.go

```diff
@@ -238,10 +240,8 @@ func (p *Processor) Update(args component.Arguments) error {
 	}
 
 	// Schedule the components to run once our component is running.
-	p.consumer.Pause()
 	p.consumer.SetConsumers(tracesProcessor, metricsProcessor, logsProcessor)
 	p.sched.Schedule(p.ctx, host, components...)
```

Comment:
there is a race condition here on Update because the consumer is not paused while we set new unstarted otel components to it. Technically it's possible that a call to consume happens right after, before the start of the scheduling